index.js

```diff
@@ -1,14 +1,23 @@
 'use strict';
+
 require('harmony-reflect');
 
-module.exports = function (obj) {
-	return new Proxy(obj, {
-		get: function (target, name, receiver) {
+module.exports = function (target) {
+	var handler = {
+		get: function (target, name) {
 			if (!(name in target)) {
 				throw new TypeError('Unknown property: ' + name);
 			}
 
-			return Reflect.get(target, name, receiver);
+			return Reflect.get(target, name);
 		}
-	});
+	};
+
+	var instance = new Proxy(target, handler);
+
+	instance.decorateWith = function (fn) {
```

Comment:
This shouldn't be added to the instance, but rather be a separate export. Please read the spec on how decorators work.

It should be used like this on a plain class:

```js
@known.decorator
class Foobar {
    baz = 1;
    someMethod() {}
}
```